core/context/context.go

```diff
@@ -0,0 +1,14 @@
+package appmodule
+
+// ExecMode defines the execution mode which can be set on a Context.
+type ExecMode uint8
+
+// All possible execution modes.
+const (
+	ExecModeCheck ExecMode = iota
```

Comment:
personal preference: reserve 0 for undefined so that you can check for that if needed and default to "ExecModeCheck"

Reply:
this was to keep inline with existing types. will check if its safe to break

Reply:
did that same as context, one less sdk dep throughout for baseapp